proto/info/edge_node_cluster.proto

```diff
@@ -0,0 +1,236 @@
+  VOLUME_STATUS_DETACHING = 5;
+  VOLUME_STATUS_DELETING = 6;
+}
+
+enum StorageVolumeRobustness {
+  VOLUME_ROBUSTNESS_UNKNOWN = 0;
+  VOLUME_ROBUSTNESS_HEALTHY = 1;
+  VOLUME_ROBUSTNESS_DEGRADED = 2;
+  VOLUME_ROBUSTNESS_FAULTED = 3;
+}
+
+enum StorageVolumePVCStatus {
+  VOLUME_PVC_STATUS_UNKNOWN = 0;
+  VOLUME_PVC_STATUS_BOUND = 1;
+  VOLUME_PVC_STATUS_PENDING = 2;
+  VOLUME_PVC_STATUS_AVAILABLE = 3;
+  VOLUME_PVC_STATUS_RELEASED = 4;
+  VOLUME_PVC_STATUS_FAILED = 5;
+}
+
+enum StorageVolumeReplicaStatus {
+  VOLUME_REPLICA_STATUS_UNKNOWN = 0;
+  VOLUME_REPLICA_STATUS_REBUILDING = 1;
+  VOLUME_REPLICA_STATUS_ONLINE = 2;
+  VOLUME_REPLICA_STATUS_FAILED = 3;
+}
+
+message KubePodNameSpaceInfo {
+  // Name of the namespace
+  string name = 1;
+
+  // Number of pods in the namespace
+  uint32 podCount = 2;
+
+  // Number of pods in the namespace that are running
+  uint32 podRunningCount = 3;
+
+  // Number of pods in the namespace that are pending
+  uint32 podPendingCount = 4;
+
+  // Number of pods in the namespace that are failed
+  uint32 podFailedCount = 5;
+
+  // Number of pods in the namespace that are succeeded
+  uint32 podSucceededCount = 6;
+}
+
+enum KubePodStatus {
+  POD_STATUS_UNKNOWN = 0;
+  POD_STATUS_PENDING = 1;
+  POD_STATUS_RUNNING = 2;
+  POD_STATUS_SUCCEEDED = 3;
+  POD_STATUS_CONTAINER_CREATING = 4;
+  POD_STATUS_CRASHLOOP_BACKOFF = 5;
+  POD_STATUS_ERROR = 6;
+  POD_STATUS_EVICTED = 7;
+  POD_STATUS_FAILED = 8;
+}
+
+message KubeEVEAppPodInfo {
+  // Name of the EVE application
+  string name = 1;
+
+  // Application Status
+  KubePodStatus status = 2;
+
+  // Restart count of the application
+  uint32 restartCount = 3;
+
+  // Restart time of the application, seconds ago
+  uint64 restartTime = 4;
+
+  // Age of the application, seconds ago
+  uint64 age = 5;
+
+  // IP address of the application, on cni0 interface
+  string ipAddress = 6;
+
+  // Node name on which the application is running
+  string nodeName = 7;
+}
+
+message KubeVolumeReplicaInfo {
+  // Name of the volume replica
+  string name = 1;
+
+  // Rebuild progress of the volume replica
+  uint32 rebuildProgressPercentage = 2;
+
+  // Replica status
+  StorageVolumeReplicaStatus status = 3;
+}
+
+message KubeVolumeInfo {
+  // Name of the volume
+  string name = 1;
+
+  // Status of the volume
+  StorageVolumeState state = 2;
+
+  // Robustness of the volume
+  StorageVolumeRobustness robustness = 3;
+
+  // Age of the volume, seconds ago
+  uint64 age = 4;
+
+  // Provisioned size of the volume in bytes
+  uint64 provisionedBytes = 5;
+
+  // Allocated size of the volume in bytes
+  uint64 allocatedBytes = 6;
+
+  // PV/PVC status of the volume
+  StorageVolumePVCStatus pvcStatus = 7;
+
+  // Replicas of the volume
+  repeated KubeVolumeReplicaInfo replica = 8;
+}
+
+message KubeStorageInfo {
+  // Overall status of Longhorn
+  StorageHealthStatus health = 1;
+
+  // time of the health status transition, seconds ago
+  uint64 transitionTime = 2;
```

Comment:
Again I think that absolute timestamp would be better given that this is published to a remote controller.